python/cudf/cudf/utils/dtypes.py

```diff
@@ -465,3 +467,137 @@ def _get_nan_for_dtype(dtype):
         return dtype.type("nan")
     else:
         return np.float64("nan")
+
+
+def get_allowed_combinations_for_operator(dtype_l, dtype_r, op):
+
+    to_numpy_ops = {
+        "__add__": _ADD_TYPES,
+        "__sub__": _SUB_TYPES,
+        "__mul__": _MUL_TYPES,
+        "__floordiv__": _FLOORDIV_TYPES,
+        "__truediv__": _TRUEDIV_TYPES,
+        "__mod__": _MOD_TYPES,
+        "__pow__": _POW_TYPES,
+    }
+    allowed = to_numpy_ops.get(op, op)
+
+    # special rules for string
+    if dtype_l == "object" or dtype_r == "object":
+        if (dtype_l == dtype_r == "object") and op == np.add:
+            return True
+        else:
+            return False
+
+    # Check if we can directly operate
+
+    for valid_combo in allowed:
+        ltype, rtype, outtype = valid_combo
+        if np.can_cast(dtype_l.char, ltype) and np.can_cast(
+            dtype_r.char, rtype
+        ):
+            return outtype
+
+    raise TypeError(
+        f"{op} not supported between " f"{dtype_l} and {dtype_r} scalars"
+    )
+
+
+_ADD_TYPES = [
```

Comment:
Can we add a comment above these indicating what they are and what they're used for.